src/frontend/src/app/pages/proposal-list/proposal-list.component.ts

```diff
@@ -0,0 +1,342 @@
+    KeyColComponent,
+    ValueColComponent,
+    FormFieldComponent,
+    InputDirective,
+    LabelDirective,
+    FormatDatePipe,
+    RouterLink,
+    CardComponent,
+  ],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  styles: [
+    `
+      @import '@cg/styles/common';
+
+      :host {
+        @include page-content;
+      }
+
+      .proposal {
+        margin-bottom: size(6);
+      }
+
+      .proposal__title,
+      .proposal__proposer {
+        word-break: break-word;
+      }
+
+      .proposal__link {
+        margin-right: size(4);
+      }
+
+      .proposal__vote {
+        font-weight: bold;
+      }
+
+      .proposal__vote--adopt {
+        color: $success;
+      }
+
+      .proposal__vote--reject {
+        color: $error;
+      }
+
+      .filter {
+        display: flex;
+        flex-direction: column;
+        margin-bottom: size(4);
+
+        @include sm {
+          flex-direction: row;
+        }
+      }
+
+      .filter__label {
+        margin-bottom: size(2);
+        margin-right: size(2);
+        width: 100%;
+
+        @include sm {
+          width: 50%;
+        }
+      }
+    `,
+  ],
+  template: `
+    <h1 class="h3">Proposals</h1>
+    <form [formGroup]="filterForm">
+      <div class="filter">
+        <div class="filter__label">Select code review status:</div>
+        <app-form-field class="filter__value">
+          <div class="radio-group">
+            @for (key of filterOptionsKeys; track key) {
+              <input
+                appInput
+                [id]="key"
+                [value]="key"
+                type="radio"
+                formControlName="filter"
+                (change)="updateFilter()"
+              />
+              <label appLabel [for]="key">
+                {{ filterOptions[key].valueOf() }}
+              </label>
+            }
+          </div>
+        </app-form-field>
+      </div>
+    </form>
+
+    @if (proposalList$ | async; as proposalList) {
+      @for (proposal of proposalList; track proposal.id; let i = $index) {
+        <cg-card class="proposal">
+          <h2 class="h4 proposal__title" slot="cardTitle">
+            {{ proposal.title }}
+          </h2>
+
+          <div slot="cardContent">
+            <app-key-value-grid [columnNumber]="2">
+              <app-key-col [id]="'proposal-id-' + i">ID</app-key-col>
+              <app-value-col [attr.aria-labelledby]="'proposal-id-' + i">
+                <a
+                  href="{{ linkBaseUrl.Proposal }}{{ proposal.id }}"
+                  target="_blank"
+                  rel="nofollow noreferrer"
+                >
+                  {{ proposal.id }}
+                </a>
+              </app-value-col>
+
+              <app-key-col [id]="'proposal-links-' + i">
+                Voting links
+              </app-key-col>
+              <app-value-col [attr.aria-labelledby]="'proposal-links-' + i">
+                @if (proposal.proposalLinks.length > 0) {
```

Comment:
There's no need to check the length of an array before looping through it. The loop will run 0 times if there are 0 elements in the array.